qiskit/visualization/utils.py

```diff
@@ -146,6 +146,61 @@ def get_param_str(op, drawer, ndigits=3):
     return param_str
 
 
+def get_bit_label(drawer, register, index, qubit=True, layout=None, cregbundle=True):
+    """Get the bit labels to display to the left of the wires."""
+    index_str = f"{index}" if drawer == "text" else f"{{{index}}}"
+
+    if register is not None:
+        if drawer == "text":
+            reg_name = f"{register.name}"
+            reg_name_index = f"{register.name}_{index}"
+        else:
+            reg_name = f"{{{register.name}}}"
+            reg_name_index = f"{{{register.name}}}_{{{index}}}"
+
+        if not qubit and cregbundle:
+            bit_label = f"{register.name}"
+            return bit_label
+
+        if register.size > 1:
```

Comment:
This big nested if statement is kind of ugly but I understand it's partly because of messiness from before. At some point I think this should be refactored, but perhaps that is beyond the scope of this PR

Reply:
Couldn't help myself. Refactored in https://github.com/Qiskit/qiskit-terra/pull/6942/commits/8a7c24a01ea3806e4e93c0d47ad0171101130b4c. Let me know what you think.